### PR TITLE
[FLINK-21348] Add tests for AdaptiveScheduler's StateWithExecutionGraph

### PR DESCRIPTION
**Pull request by rmetzger**
## What is the purpose of the change

Add more tests for StateWithExecutionGraph

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraphTest.java [hunk 1]
```diff
@@ -75,34 +82,112 @@ public void testSuspendCanBeCalledWhenExecutionGraphHasReachedGloballyTerminalSt
...
+
+    private TestingStateWithExecutionGraph createStateWithExecutionGraph(
+            MockStateWithExecutionGraphContext context, ExecutionGraph executionGraph) {
+        final OperatorCoordinatorHandler operatorCoordinatorHandler =
+                new OperatorCoordinatorHandlerImplementation(
+                        executionGraph,
+                        globalFailure -> {
+                            throw new FlinkRuntimeException(
+                                    "No global failures are expected", globalFailure);
+                        });
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraphTest.java:168]: Could we use the `TestingOperatorCoordinatorHandler` here?

**Review by tillrohrmann — COMMENTED** [review]
Thanks for creating this PR @rmetzger. The changes look overall good. I had few comments. Please take a look.

**Comment by rmetzger** [comment]
Thanks a lot for your fast feedback! I hope I addressed all your concerns!

**Review by tillrohrmann — APPROVED** [review]
Thanks for updating this PR @rmetzger. LGTM. +1 for merging after AZP gives green light.

**Comment by rmetzger** [comment]
Thanks a lot for your review. I'll merge this soon ;)

**Pull request closed without merging**